File: src/main/java/net/yakclient/mixin/internal/methodadapter/tree/TreeBIPatternMatcher.java
package net.yakclient.mixin.internal.methodadapter.tree;

import net.yakclient.mixin.internal.instruction.ASMInstruction;
import net.yakclient.mixin.internal.instruction.Instruction;
import org.objectweb.asm.tree.InsnList;

public class TreeBIPatternMatcher extends TreeMixinPatternMatcher {
    public TreeBIPatternMatcher(Instruction instructions) {
        super(instructions);
    }

    @Override
    public void transform(InsnList insn) {

    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/methodadapter/tree/TreeBIPatternMatcher.kt
package net.yakclient.mixins.base.internal.methodadapter.tree

import net.yakclient.mixins.base.internal.instruction.Instruction
import org.objectweb.asm.tree.InsnList
import org.objectweb.asm.tree.MethodInsnNode

/**
 * Defines the pattern for inserting mixins before any method invokes.
 *
 *
 * Type: Before invoke
 *
 * @author Durgan McBroom
 */
class TreeBIPatternMatcher(
    instructions: Instruction
) : TreeMixinPatternMatcher(instructions) {
    override fun transform(insn: InsnList) {
        for (node in insn) {
            if (node is MethodInsnNode) this.insertInsn(node, insn)
        }
    }
}